### PR TITLE
Upgrade to Sentry SDK 6

### PR DESCRIPTION
**Pull request by verglor**

### CODE REVIEW

File: src/main/groovy/grails/plugin/sentry/SpringSecuritySentryUserProvider.groovy [hunk 1]
```diff
@@ -37,57 +33,45 @@ class SpringSecurityUserEventProcessor implements EventProcessor {
...
-                        usernamePropertyName = config.springSecurityUserProperties.username
-                    }
-                    if (config.springSecurityUserProperties.data) {
-                        data = config.springSecurityUserProperties.data
-                    }
-                }
+                String idPropertyName = config.springSecurityUserProperties?.id ?: 'id'
+                String emailPropertyName = config.springSecurityUserProperties?.email
+                String usernamePropertyName = config.springSecurityUserProperties?.username ?: 'username'
+                List<String> data = config.springSecurityUserProperties?.data ?: []
```
Comment (YaeriusFCM) [inline src/main/groovy/grails/plugin/sentry/SpringSecuritySentryUserProvider.groovy:56]: I ❤️ this code reduction 😃

**Review by YaeriusFCM — APPROVED** [review]

File: src/main/groovy/grails/plugin/sentry/SpringSecurityUserEventProcessor.groovy [hunk 2]
```diff
@@ -85,9 +96,9 @@ class SpringSecurityUserEventProcessor implements EventProcessor {
 
                 User user = new User(id: id, username: username, ipAddress: ipAddress, email: email, unknown: extraData)
                 event.setUser(user)
-                return event
             }
         }
+        return event
```
Comment (leni-kirilov) [inline src/main/groovy/grails/plugin/sentry/SpringSecurityUserEventProcessor.groovy]: no really need for "return" word here

File: grails-app/controllers/grails/plugin/sentry/SentryControllerInterceptor.groovy [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    boolean before() {
+        request[SENTRY_CONTROLLER_SPAN] = Sentry.span?.startChild('grails.controller', "${controllerName}/${actionName}")
+        true
+    }
+
+    boolean after() {
+        if (request[SENTRY_CONTROLLER_SPAN] instanceof ISpan) {
+            (request[SENTRY_CONTROLLER_SPAN] as ISpan).finish(SpanStatus.OK)
+        }
+        request[SENTRY_VIEW_SPAN] = Sentry.span?.startChild('grails.view', view)
```
Comment (leni-kirilov) [inline grails-app/controllers/grails/plugin/sentry/SentryControllerInterceptor.groovy:27]: Does this make the assumption there is always a `grails.view` after a controller call? What if the user of the plugin is actually building an API which doesn't call a .gsp or .gson or .gml? 

I found a similar construct here : `com.adverity.interceptors.LoggingInterceptor#afterView`

I'm not sure I agree with this change unless I'm misinterpreting it

File: src/main/groovy/grails/plugin/sentry/SentryGrailsPlugin.groovy [hunk 4]
```diff
@@ -135,8 +147,9 @@ class SentryGrailsPlugin extends Plugin {
             appender.start()
         }
 
-        SentryOptions options = new SentryOptions()
+        SentryOptions options = applicationContext.getBean(SentryOptions)
         options.enableExternalConfiguration = true
+        options.sendDefaultPii = true
```
Comment (leni-kirilov) [inline src/main/groovy/grails/plugin/sentry/SentryGrailsPlugin.groovy:154]: According to this https://develop.sentry.dev/sdk/data-handling/ - you are enabling the sending of sensitive data such as authorizations and cookies. 
Why do we need this? The user email/name?

File: src/main/groovy/grails/plugin/sentry/SpringSecuritySentryUserProvider.groovy (hunk 1 above)
Comment (leni-kirilov) [reply src/main/groovy/grails/plugin/sentry/SpringSecuritySentryUserProvider.groovy:56]: image

**Review by leni-kirilov — COMMENTED** [review]
A few questions from my side.

Also - do you intend to add some unit tests for these changes?

File: src/main/groovy/grails/plugin/sentry/SentryGrailsPlugin.groovy (hunk 4 above)
Comment (verglor) [reply src/main/groovy/grails/plugin/sentry/SentryGrailsPlugin.groovy:154]: Exactly - it's for user info.

File: src/main/groovy/grails/plugin/sentry/SpringSecurityUserEventProcessor.groovy (hunk 2 above)
Comment (verglor) [reply src/main/groovy/grails/plugin/sentry/SpringSecurityUserEventProcessor.groovy]: It's optional but it's more descriptive to have explicit return keyword.

File: grails-app/controllers/grails/plugin/sentry/SentryControllerInterceptor.groovy (hunk 3 above)
Comment (verglor) [reply grails-app/controllers/grails/plugin/sentry/SentryControllerInterceptor.groovy:27]: Good point. I have added condition there: a3fae414

**Pull request merged**